cli/src/java/org/apache/hadoop/hive/cli/CliDriver.java

```diff
@@ -445,6 +445,17 @@ public void handle(Signal signal) {
         }
         break;
       case '"':
+        if (!escape) {
```

Comment:
How is this different to the case below for '\''? Maybe I'm missing something but I can't see a difference and the current code has both cases dropping to the same block so the end effect would be the same surely?

Reply:
> How is this different to the case below for `'\''`? Maybe I'm missing something but I can't see a difference and the current code has both cases dropping to the same block so the end effect would be the same surely?

There is not different to the case below for '\''. 
Because single quotes and double quotes work the same in a sql statement.

Reply:
If it's not different then why introduce duplicated code for both cases? Surely the way it was previously both cases would drop down to the same code block which is effectively the same as this?

Reply:
> If it's not different then why introduce duplicated code for both cases? Surely the way it was previously both cases would drop down to the same code block which is effectively the same as this?

Oh sorry,  you are right. I found this case in spark.
Spark dependent on hive-1.2.1 which  process these case incorrectly.
So I override the splitSemicolon method in SparkSQLCLIDriver with scala. 
However, scala is different to java. 
Scala does not need break to break case.
Thanks, I will close this PR.

Reply:
:+1: